### ISSUE TITLE
Shards keep reconnecting (why is that happening?)

### ISSUE THREAD

**Issue by AbelCode1** [brblacky/lavamusic#70]

**Comment by sandarutharuneth** [comment]
Same as mine shards are reconnecting

**Comment by brblacky** [comment]
Please ask in discord dev server there help us

**Comment by AbelCode1** [comment]
I think shards are reconnecting because you have to choose between erela or discord.js. 
either full erela or full discord.js

my other thought of shards reconnecting is because the database (maybe try using a different database provider?)

**Comment by Jeve-Stobs** [comment]
**TL;DR:** It's expected behaviour

Discord requests your bot to be disconnected randomly every 15 minutes to 4 hours, but it should automatically resume. 

The bot shouldn't actually restart, it should just reconnect to the gateway. 
It doesn't add to any rate limit or connection limit and you shouldn't experience this very often if you have one shard or fewer.

**Comment by AbelCode1** [comment]
> **TL;DR:** It's expected behaviour
> 
> Discord requests your bot to be disconnected randomly every 15 minutes to 4 hours, but it should automatically resume.
> 
> The bot shouldn't actually restart, it should just reconnect to the gateway. It doesn't add to any rate limit or connection limit and you shouldn't experience this very often if you have one shard or fewer.

I experience this very often like very very often. I only have 1 shard the shard name is Shard#0

**Comment by Jeve-Stobs** [comment]
That's still normal. I have a bot on two shards and it does the same thing. Discord just wants to make sure that your bot is alive. If it was spamming reconnects, then you would have to look at your code. I would recommend only using the webhook when the reconnection lasts for >30 sec.
